docs/transformers/external/starlark.md

```diff
@@ -12,4 +12,76 @@ nav_order: 1
+
+Second argument which is optional is the validator function. If the input from the user does not comply with the provided validator function then question is asked again until the expected input is received.
+
+Example usage can be found [here](https://github.com/konveyor/move2kube-transformers/blob/2a36d11efb996f2f38cdb3e3ecccfe9378e3a444/cloud-foundry-to-ce-iks-roks/cedockerfile/cedockerfile.star#L52-L63).
+
+
+### File system
+Move2Kube provides a set a functions in starlark for filesystem based operations.
+
+| Function name in starlark | Purpose | Arguments | Returns |
+| ----------- | ----------- | ----------- | ----------- | 
+| `exists`| Check if the provided file exists or not | filepath as string| boolean |
+| `read`| Reads the given file | filepath as string | string |
+| `read_dir`| Obtain list of files names in a directory | directory path as string | []string | 
+| `is_dir`| Check if the given path is a directory or not | path as string | boolean | 
+| `get_yamls_with_type_meta`| Return files by yaml kind | path to directory containing yamls as string <br/> kind of yamls required as string | []string| 
+| `find_xml_path`|  |  | | 
+| `get_files_with_pattern` | Get files in the given directory with a particular extension | path to directory as string <br/> list of extensions as []string | []string |
+| `path_join` | Join two paths | first as path as string <br/> second path as string| string |
+| `write` | Write data to a file and returns number of bytes written | path to a file as string <br/> data to be written as a string <br/> file permissions (optional) as an integer | int
+| `path_base` | Return last element of the given path | path as string | string
+| `path_rel` | Return relative path | base path as string <br/> target path as string | string
+
+### Cryptography
+Move2Kube provides a set a functions in starlark for Cryptography.
+
+| Function name in starlark | Purpose | Arguments | Returns |
+| ----------- | ----------- | ----------- | ----------- | 
+| `enc_aes_cbc_pbkdf`| Encrypt the data using AES 256 CBC mode with Pbkdf key derivation and return the cipher text | key as string <br /> data as string | string |
+| `enc_rsa_cert`| Encrypt the data using RSA PKCS1v15 algorithm with certificate as key and return cipher text | certificate as string <br /> data as string | string |
+
+
+### Archival
+Move2Kube provides a set a functions in starlark for Archival.
+
+| Function name in starlark | Purpose | Arguments | Returns |
+| ----------- | ----------- | ----------- | ----------- | 
+| `arch_tar_gzip_str`| archive a set of files and compression using gzip and return the digest | path to the directory containing the files as string | string |
+| `arch_tar_str`| archive a set of files and compression without compression and and return the digest | path to the directory containing files as string | string |
+
+
+## Variables
+Move2Kube provides a set of variables that can be used in the starlark transformer code.
```

Comment:
Would be good to mention that these variables are for exposing values required by the transformer.